### PR TITLE
Feat 8 user profile

### PR DESCRIPTION
**Pull request by SinchanaMS**
This request adds the following features:

1. User Profile - The user can navigate to their profile by clicking on the user icon on the top-right.
2. The User profile card gives the stats of the no. of videos liked, saved in the watchlist and the no. of playlists created.

Please review the following files:
1. [Header.jsx](https://github.com/SinchanaMS/react-video-library/pull/10/files#diff-8c468c347b974539dbc7411567675397475e1aeac509adf899831746084c1179)
2. [UserProfileCard.jsx](https://github.com/SinchanaMS/react-video-library/pull/10/files#diff-4b38b8947fddc76aebe98c14d89ff1bd1228706ade3fb1731e92e649ee47f2eb)
3. [UserProfile.jsx](https://github.com/SinchanaMS/react-video-library/pull/10/files#diff-4d270b721a1263274f8b3369d64ba1a0eac7e8077fef08498613b209aa2973f3)
4. [profile.css](https://github.com/SinchanaMS/react-video-library/pull/10/files#diff-7954008ca40022dae736393e435386045f4f124edf6887135383ee1ed4cd0568)

The rest can be ignored as it was all cleanup of repetitive code.

### CODE REVIEW

File: src/components/UserProfileCard.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useVideo } from "contexts/contexts";
+
+export default function UserProfileCard() {
+  const { videoData } = useVideo();
+  const { watchList, likedList, playlists } = videoData;
+  console.log(videoData, watchList, likedList, playlists);
```
Comment (muskanbagrecha) [inline src/components/UserProfileCard.jsx]: Remove console.log

File: src/components/UserProfileCard.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  return (
+    <div className="profile-card shadow">
+      <header className="profile-card-header">
+        <h1>Hello, Daniel!</h1>
```
Comment (muskanbagrecha) [inline src/components/UserProfileCard.jsx]: Is this meant to be hardcoded?

File: src/contexts/AuthContext.jsx [hunk 3]
```diff
@@ -8,7 +8,10 @@ const AuthProvider = ({ children }) => {
   const [isLoggedIn, setIsLoggedIn] = useState(false);
   const userToken = localStorage.getItem("userToken");
   const navigate = useNavigate();
-
+  const testUser = {
+    email: "danielFelton@gmail.com",
+    password: "FortunaMajor",
+  };
   useEffect(() => {
```
Comment (muskanbagrecha) [inline src/contexts/AuthContext.jsx:18]: Please add dep.

File: src/pages/Authentication/Login.jsx [hunk 4]
```diff
@@ -27,7 +22,7 @@ export default function Login() {
     e.preventDefault();
     sendLoginData(loginData);
   };
-
+  console.log(lastLocation);
```
Comment (muskanbagrecha) [inline src/pages/Authentication/Login.jsx]: Remove console.log

**Review by muskanbagrecha — APPROVED** [review]
Code looks clean. Love the UI. Could not find much issues.

File: src/components/UserProfileCard.jsx (hunk 2 above)
Comment (SinchanaMS) [reply src/components/UserProfileCard.jsx]: No, that was a miss. Thank you for pointing it out!

File: src/App.js [hunk 5]
```diff
@@ -4,11 +4,13 @@ import Header from "components/Header";
...
 import { Toaster } from "react-hot-toast";
+import { useTheme } from "contexts/contexts";
 
 function App() {
   const location = useLocation();
+  const { theme } = useTheme();
   return (
-    <div className="App">
+    <div className={theme === "light" ? "App light" : "App dark"}>
       {location.pathname !== "/login" && location.pathname !== "/signup" && (
```
Comment (pankaj-prog) [inline src/App.js:14]: you could also do .. location.pathname !== "/login" || "/signup"

File: src/components/Header.jsx [hunk 6]
```diff
@@ -30,90 +30,63 @@ export default function Header() {
...
-              </p>
+              <Link to="/profile" className="user-actions">
+                Profile
+              </Link>
             </div>
+            {isLoggedIn ? (
+              <div className="user-options-ctas">
+                <Link
+                  to="/"
+                  onClick={() => logoutHandler()}
```
Comment (pankaj-prog) [inline src/components/Header.jsx:68]: you can directly right logoutHandler

File: src/components/PlaylistModal.jsx [hunk 7]
```diff
@@ -44,7 +41,7 @@ export default function PlaylistModal({ video }) {
               onClick={() =>
                 videoInPlaylist(playlist)
                   ? toast.error("Video already exists in the playlist")
-                  : addToPlaylist(playlist._id, video, videoDispatch)
+                  : addToPlaylist(playlist._id, video, userToken, videoDispatch)
```
Comment (pankaj-prog) [inline src/components/PlaylistModal.jsx:44]: suggestion :  import this usertoken in the addToPlaylist function file only

**Review by pankaj-prog — APPROVED** [review]

File: src/components/PlaylistModal.jsx (hunk 7 above)
Comment (SinchanaMS) [reply src/components/PlaylistModal.jsx:44]: That was what I did initially, but it wasn't capturing the userToken unless passed as a prop. I plan to look into the reason and change it later on, thank you!

**Comment by SinchanaMS** [comment]
Thank you for the reviews, merging to dev.

**Pull request merged**